accepted/2024/[TS]PMUI-Readme-rendering.md

```diff
@@ -0,0 +1,117 @@
+
+Currently the NuGet Server API has the a property called ReadmeUrl as part of the [Package metadata](https://learn.microsoft.com/en-us/nuget/api/registration-base-url-resource#catalog-entry).
+This is a link to the readme which is being rendered in nuget.org.
+For rendering the ReadMe in the PM UI we need to use the raw markdown.
+Currently the only way to get this markdown requires downloading the nupkg to search inside of it and determine if there is a ReadMe available.
+We want to avoid exploratory nupkg downloads as this can effect performance. We want the API to expose a new property which provides a link to download the markdown.
+
+## Explanation
+
+### Functional explanation
+#### PM UI
+When a package is selected we will determine if a ReadMe file exists and if it does we'll render it in the PM UI.
+
+The PM UI will be updated to have tabs for the Package Details and the the ReadMe.
+This UX will be displayed for both the Browse and installed tabs.
+It will also be displayed for both the solution level and project level package managers.
+![Alt text](https://github.com/NuGet/Home/assets/89422562/81b24877-f12f-4783-905c-4a155d3c7693)
+
+The ReadMe file will be selected from disk if available.
+
+#### Nuget Server API
+Package metadata Catalog Entry will be updated to include a new property, ReadmeFileUrl, which will contain a URL that can be used to download the Readme. 
+The URL will only point to the embedded readme for the current package version.
+If no readme is available nothing is returned. 
+
+### Technical explanation
+
+#### PM UI
+##### Rendering Markdown
+We want to leverage the [IMarkdownPreview](https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=/src/Productivity/MarkdownLanguageService/Impl/Markdown.Platform/Preview/IMarkdownPreview.cs) class to render the ReadMe in the IDE.
+A new instance can be created using [PreviewBuilder](https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=/src/Productivity/MarkdownLanguageService/Impl/Markdown.Platform/Preview/PreviewBuilder.cs).
+
+We can use the Preview builder as follows:
+```C#
+//This creates a new instance of the preview builder
+var markdownPreview = new PreviewBuilder().Build();
+
+//We update the current markdown being rendered by calling "UpdateContentAsync"
+markdownPreview.UpdateContentAsync(markDown ?? string.Empty, ScrollHint.None)
+
+//IMarkdownPreview.VisualElement contains the FrameworkElement to be passed to the view
+MarkdownPreviewControl = markdownPreview.VisualElement
+```
+##### Locating the ReadMe
+We will focus the MVP on loading the ReadMe only for packages which are already downloaded.
+
+Will use ZipArchive to locate the nuspec within the Nupkg.
+Then use NuspecReader to locate the ReadMe within the NupKg, and extract it using ZipArchive.
```

Comment:
README file is already extracted to the global packages folder, correct? I am trying to understand the reasoning behind using NuspecReader to locate the file within the nupkg. Please let me know if I am missing something.

<img width="443" alt="image" src="https://github.com/NuGet/Home/assets/52756182/666f0cba-04e9-4171-bdd8-2f017b310458">